diff -up ModemManager-1.5.991/plugins/huawei/mm-modem-helpers-huawei.c.than ModemManager-1.5.991/plugins/huawei/mm-modem-helpers-huawei.c
--- ModemManager-1.5.991/plugins/huawei/mm-modem-helpers-huawei.c.than	2016-04-12 14:03:15.519328232 +0200
+++ ModemManager-1.5.991/plugins/huawei/mm-modem-helpers-huawei.c	2016-04-12 13:56:58.978491128 +0200
@@ -190,7 +190,11 @@ match_info_to_ip4_addr (GMatchInfo *matc
     if (!bin || bin_len != 4)
         goto done;
 
-    *out_addr = GUINT32_TO_BE (*((guint32 *) bin));
+#if __BYTE_ORDER == __BIG_ENDIAN
+    *out_addr = GUINT32_TO_LE (*((guint32 *) bin));
+#else
+    *out_addr = GUINT32_TO_BE (*((guint32 *) bin));   
+#endif
     success = TRUE;
 
 done:
